## Importing the packages.

In [1]:
import pandas as pd
import numpy as np 

import matplotlib.pyplot as plt
%matplotlib inline

#from resnets_utils import *
from sklearn.model_selection import train_test_split

from keras import backend as k
from keras import regularizers
from keras import applications
from keras.preprocessing import image
from keras import optimizers
from keras.optimizers import SGD
from keras.models import Sequential ,Model, load_model
from keras.layers import Dense, Dropout, Flatten, Conv2D,MaxPooling2D, MaxPool2D, GlobalAveragePooling2D, Input, LSTM, TimeDistributed, Activation
from keras.utils import to_categorical
from keras.callbacks import ModelCheckpoint, EarlyStopping, LearningRateScheduler
from keras.preprocessing.image import ImageDataGenerator 

import random
from PIL import Image

import os 
import cv2
import imutils


Using TensorFlow backend.


## Reading the csv and loading the data.

dataset = pd.read_csv("train_labels.csv")

dataset.head(5)

Defining the list to hold data.

In [2]:
D = []
model_history = []

Using open CV for data augmentation . Here we are generating the data by rotating the image in
certain angle. currently creating image at every 90 degree.

Loading data from the TRAIN folder.

Under the train folder we have two folder named as '1' for parasite or '0' for uninfected.

In [3]:
infected_dir = 'train/1'
uninfected_dir = 'train/0'

INFECTED DATA

In [4]:
labels = os.listdir(infected_dir)

for label in labels:
    
    filename = infected_dir +'/'+label
    
    rot_img = image.load_img(filename, target_size=(128,128,3), color_mode="rgb")

    img = image.img_to_array(rot_img)
    img = img/255    
    D.append((img, 1))

    print("Image : "+ filename)

Image : train/1/00000.jpg
Image : train/1/00001.jpg
Image : train/1/00002.jpg
Image : train/1/00005.jpg
Image : train/1/00007.jpg
Image : train/1/00008.jpg
Image : train/1/00010.jpg
Image : train/1/00011.jpg
Image : train/1/00012.jpg
Image : train/1/00013.jpg
Image : train/1/00014.jpg
Image : train/1/00015.jpg
Image : train/1/00016.jpg
Image : train/1/00020.jpg
Image : train/1/00025.jpg
Image : train/1/00026.jpg
Image : train/1/00027.jpg
Image : train/1/00031.jpg
Image : train/1/00034.jpg
Image : train/1/00035.jpg
Image : train/1/00036.jpg
Image : train/1/00037.jpg
Image : train/1/00039.jpg
Image : train/1/00040.jpg
Image : train/1/00041.jpg
Image : train/1/00043.jpg
Image : train/1/00046.jpg
Image : train/1/00051.jpg
Image : train/1/00054.jpg
Image : train/1/00056.jpg
Image : train/1/00059.jpg
Image : train/1/00061.jpg
Image : train/1/00062.jpg
Image : train/1/00064.jpg
Image : train/1/00068.jpg
Image : train/1/00070.jpg
Image : train/1/00071.jpg
Image : train/1/00072.jpg
Image : trai

Image : train/1/00770.jpg
Image : train/1/00771.jpg
Image : train/1/00778.jpg
Image : train/1/00780.jpg
Image : train/1/00787.jpg
Image : train/1/00790.jpg
Image : train/1/00791.jpg
Image : train/1/00794.jpg
Image : train/1/00796.jpg
Image : train/1/00799.jpg
Image : train/1/00800.jpg
Image : train/1/00803.jpg
Image : train/1/00805.jpg
Image : train/1/00807.jpg
Image : train/1/00809.jpg
Image : train/1/00813.jpg
Image : train/1/00814.jpg
Image : train/1/00819.jpg
Image : train/1/00821.jpg
Image : train/1/00822.jpg
Image : train/1/00823.jpg
Image : train/1/00826.jpg
Image : train/1/00827.jpg
Image : train/1/00828.jpg
Image : train/1/00829.jpg
Image : train/1/00832.jpg
Image : train/1/00833.jpg
Image : train/1/00835.jpg
Image : train/1/00837.jpg
Image : train/1/00838.jpg
Image : train/1/00840.jpg
Image : train/1/00843.jpg
Image : train/1/00844.jpg
Image : train/1/00846.jpg
Image : train/1/00848.jpg
Image : train/1/00850.jpg
Image : train/1/00852.jpg
Image : train/1/00856.jpg
Image : trai

Image : train/1/01564.jpg
Image : train/1/01567.jpg
Image : train/1/01568.jpg
Image : train/1/01569.jpg
Image : train/1/01571.jpg
Image : train/1/01572.jpg
Image : train/1/01573.jpg
Image : train/1/01576.jpg
Image : train/1/01577.jpg
Image : train/1/01578.jpg
Image : train/1/01580.jpg
Image : train/1/01584.jpg
Image : train/1/01585.jpg
Image : train/1/01589.jpg
Image : train/1/01590.jpg
Image : train/1/01594.jpg
Image : train/1/01595.jpg
Image : train/1/01596.jpg
Image : train/1/01599.jpg
Image : train/1/01600.jpg
Image : train/1/01601.jpg
Image : train/1/01602.jpg
Image : train/1/01604.jpg
Image : train/1/01605.jpg
Image : train/1/01608.jpg
Image : train/1/01609.jpg
Image : train/1/01610.jpg
Image : train/1/01611.jpg
Image : train/1/01613.jpg
Image : train/1/01614.jpg
Image : train/1/01615.jpg
Image : train/1/01618.jpg
Image : train/1/01619.jpg
Image : train/1/01620.jpg
Image : train/1/01621.jpg
Image : train/1/01622.jpg
Image : train/1/01624.jpg
Image : train/1/01626.jpg
Image : trai

Image : train/1/02227.jpg
Image : train/1/02229.jpg
Image : train/1/02231.jpg
Image : train/1/02233.jpg
Image : train/1/02234.jpg
Image : train/1/02235.jpg
Image : train/1/02237.jpg
Image : train/1/02240.jpg
Image : train/1/02241.jpg
Image : train/1/02243.jpg
Image : train/1/02245.jpg
Image : train/1/02246.jpg
Image : train/1/02247.jpg
Image : train/1/02251.jpg
Image : train/1/02253.jpg
Image : train/1/02255.jpg
Image : train/1/02256.jpg
Image : train/1/02260.jpg
Image : train/1/02261.jpg
Image : train/1/02262.jpg
Image : train/1/02264.jpg
Image : train/1/02266.jpg
Image : train/1/02269.jpg
Image : train/1/02272.jpg
Image : train/1/02273.jpg
Image : train/1/02274.jpg
Image : train/1/02275.jpg
Image : train/1/02277.jpg
Image : train/1/02278.jpg
Image : train/1/02284.jpg
Image : train/1/02285.jpg
Image : train/1/02286.jpg
Image : train/1/02289.jpg
Image : train/1/02291.jpg
Image : train/1/02292.jpg
Image : train/1/02293.jpg
Image : train/1/02296.jpg
Image : train/1/02297.jpg
Image : trai

Image : train/1/02881.jpg
Image : train/1/02884.jpg
Image : train/1/02891.jpg
Image : train/1/02893.jpg
Image : train/1/02896.jpg
Image : train/1/02897.jpg
Image : train/1/02898.jpg
Image : train/1/02899.jpg
Image : train/1/02900.jpg
Image : train/1/02901.jpg
Image : train/1/02909.jpg
Image : train/1/02910.jpg
Image : train/1/02911.jpg
Image : train/1/02912.jpg
Image : train/1/02914.jpg
Image : train/1/02917.jpg
Image : train/1/02919.jpg
Image : train/1/02923.jpg
Image : train/1/02924.jpg
Image : train/1/02926.jpg
Image : train/1/02927.jpg
Image : train/1/02931.jpg
Image : train/1/02934.jpg
Image : train/1/02936.jpg
Image : train/1/02938.jpg
Image : train/1/02939.jpg
Image : train/1/02943.jpg
Image : train/1/02944.jpg
Image : train/1/02946.jpg
Image : train/1/02947.jpg
Image : train/1/02948.jpg
Image : train/1/02949.jpg
Image : train/1/02950.jpg
Image : train/1/02951.jpg
Image : train/1/02957.jpg
Image : train/1/02958.jpg
Image : train/1/02960.jpg
Image : train/1/02965.jpg
Image : trai

Image : train/1/03521.jpg
Image : train/1/03523.jpg
Image : train/1/03524.jpg
Image : train/1/03525.jpg
Image : train/1/03527.jpg
Image : train/1/03530.jpg
Image : train/1/03531.jpg
Image : train/1/03532.jpg
Image : train/1/03533.jpg
Image : train/1/03534.jpg
Image : train/1/03538.jpg
Image : train/1/03540.jpg
Image : train/1/03544.jpg
Image : train/1/03546.jpg
Image : train/1/03547.jpg
Image : train/1/03548.jpg
Image : train/1/03553.jpg
Image : train/1/03554.jpg
Image : train/1/03555.jpg
Image : train/1/03556.jpg
Image : train/1/03560.jpg
Image : train/1/03562.jpg
Image : train/1/03563.jpg
Image : train/1/03564.jpg
Image : train/1/03566.jpg
Image : train/1/03568.jpg
Image : train/1/03570.jpg
Image : train/1/03572.jpg
Image : train/1/03573.jpg
Image : train/1/03574.jpg
Image : train/1/03575.jpg
Image : train/1/03576.jpg
Image : train/1/03577.jpg
Image : train/1/03578.jpg
Image : train/1/03583.jpg
Image : train/1/03584.jpg
Image : train/1/03585.jpg
Image : train/1/03586.jpg
Image : trai

Image : train/1/04199.jpg
Image : train/1/04202.jpg
Image : train/1/04203.jpg
Image : train/1/04204.jpg
Image : train/1/04207.jpg
Image : train/1/04209.jpg
Image : train/1/04214.jpg
Image : train/1/04215.jpg
Image : train/1/04216.jpg
Image : train/1/04220.jpg
Image : train/1/04228.jpg
Image : train/1/04229.jpg
Image : train/1/04231.jpg
Image : train/1/04233.jpg
Image : train/1/04234.jpg
Image : train/1/04236.jpg
Image : train/1/04239.jpg
Image : train/1/04240.jpg
Image : train/1/04242.jpg
Image : train/1/04245.jpg
Image : train/1/04246.jpg
Image : train/1/04248.jpg
Image : train/1/04249.jpg
Image : train/1/04250.jpg
Image : train/1/04251.jpg
Image : train/1/04253.jpg
Image : train/1/04258.jpg
Image : train/1/04263.jpg
Image : train/1/04264.jpg
Image : train/1/04267.jpg
Image : train/1/04270.jpg
Image : train/1/04271.jpg
Image : train/1/04272.jpg
Image : train/1/04275.jpg
Image : train/1/04277.jpg
Image : train/1/04279.jpg
Image : train/1/04281.jpg
Image : train/1/04284.jpg
Image : trai

Image : train/1/04898.jpg
Image : train/1/04906.jpg
Image : train/1/04913.jpg
Image : train/1/04914.jpg
Image : train/1/04916.jpg
Image : train/1/04920.jpg
Image : train/1/04923.jpg
Image : train/1/04926.jpg
Image : train/1/04928.jpg
Image : train/1/04929.jpg
Image : train/1/04930.jpg
Image : train/1/04934.jpg
Image : train/1/04936.jpg
Image : train/1/04937.jpg
Image : train/1/04938.jpg
Image : train/1/04939.jpg
Image : train/1/04940.jpg
Image : train/1/04943.jpg
Image : train/1/04945.jpg
Image : train/1/04946.jpg
Image : train/1/04947.jpg
Image : train/1/04950.jpg
Image : train/1/04954.jpg
Image : train/1/04956.jpg
Image : train/1/04962.jpg
Image : train/1/04963.jpg
Image : train/1/04964.jpg
Image : train/1/04968.jpg
Image : train/1/04970.jpg
Image : train/1/04974.jpg
Image : train/1/04975.jpg
Image : train/1/04980.jpg
Image : train/1/04981.jpg
Image : train/1/04982.jpg
Image : train/1/04983.jpg
Image : train/1/04988.jpg
Image : train/1/04989.jpg
Image : train/1/04991.jpg
Image : trai

Image : train/1/05521.jpg
Image : train/1/05526.jpg
Image : train/1/05529.jpg
Image : train/1/05530.jpg
Image : train/1/05531.jpg
Image : train/1/05535.jpg
Image : train/1/05536.jpg
Image : train/1/05537.jpg
Image : train/1/05539.jpg
Image : train/1/05540.jpg
Image : train/1/05541.jpg
Image : train/1/05549.jpg
Image : train/1/05551.jpg
Image : train/1/05552.jpg
Image : train/1/05553.jpg
Image : train/1/05556.jpg
Image : train/1/05557.jpg
Image : train/1/05558.jpg
Image : train/1/05560.jpg
Image : train/1/05562.jpg
Image : train/1/05566.jpg
Image : train/1/05567.jpg
Image : train/1/05569.jpg
Image : train/1/05571.jpg
Image : train/1/05574.jpg
Image : train/1/05575.jpg
Image : train/1/05576.jpg
Image : train/1/05579.jpg
Image : train/1/05580.jpg
Image : train/1/05581.jpg
Image : train/1/05585.jpg
Image : train/1/05588.jpg
Image : train/1/05590.jpg
Image : train/1/05591.jpg
Image : train/1/05593.jpg
Image : train/1/05594.jpg
Image : train/1/05595.jpg
Image : train/1/05597.jpg
Image : trai

Image : train/1/06170.jpg
Image : train/1/06172.jpg
Image : train/1/06173.jpg
Image : train/1/06174.jpg
Image : train/1/06175.jpg
Image : train/1/06176.jpg
Image : train/1/06178.jpg
Image : train/1/06179.jpg
Image : train/1/06183.jpg
Image : train/1/06186.jpg
Image : train/1/06188.jpg
Image : train/1/06189.jpg
Image : train/1/06191.jpg
Image : train/1/06193.jpg
Image : train/1/06194.jpg
Image : train/1/06197.jpg
Image : train/1/06198.jpg
Image : train/1/06202.jpg
Image : train/1/06203.jpg
Image : train/1/06206.jpg
Image : train/1/06207.jpg
Image : train/1/06208.jpg
Image : train/1/06212.jpg
Image : train/1/06213.jpg
Image : train/1/06214.jpg
Image : train/1/06215.jpg
Image : train/1/06218.jpg
Image : train/1/06219.jpg
Image : train/1/06220.jpg
Image : train/1/06221.jpg
Image : train/1/06222.jpg
Image : train/1/06223.jpg
Image : train/1/06226.jpg
Image : train/1/06230.jpg
Image : train/1/06232.jpg
Image : train/1/06233.jpg
Image : train/1/06234.jpg
Image : train/1/06238.jpg
Image : trai

Image : train/1/06896.jpg
Image : train/1/06898.jpg
Image : train/1/06899.jpg
Image : train/1/06901.jpg
Image : train/1/06902.jpg
Image : train/1/06904.jpg
Image : train/1/06906.jpg
Image : train/1/06907.jpg
Image : train/1/06908.jpg
Image : train/1/06909.jpg
Image : train/1/06912.jpg
Image : train/1/06915.jpg
Image : train/1/06916.jpg
Image : train/1/06920.jpg
Image : train/1/06921.jpg
Image : train/1/06925.jpg
Image : train/1/06928.jpg
Image : train/1/06930.jpg
Image : train/1/06933.jpg
Image : train/1/06934.jpg
Image : train/1/06935.jpg
Image : train/1/06937.jpg
Image : train/1/06939.jpg
Image : train/1/06944.jpg
Image : train/1/06949.jpg
Image : train/1/06954.jpg
Image : train/1/06955.jpg
Image : train/1/06958.jpg
Image : train/1/06961.jpg
Image : train/1/06962.jpg
Image : train/1/06964.jpg
Image : train/1/06965.jpg
Image : train/1/06969.jpg
Image : train/1/06970.jpg
Image : train/1/06974.jpg
Image : train/1/06975.jpg
Image : train/1/06977.jpg
Image : train/1/06982.jpg
Image : trai

Image : train/1/07559.jpg
Image : train/1/07572.jpg
Image : train/1/07575.jpg
Image : train/1/07576.jpg
Image : train/1/07577.jpg
Image : train/1/07578.jpg
Image : train/1/07579.jpg
Image : train/1/07580.jpg
Image : train/1/07581.jpg
Image : train/1/07582.jpg
Image : train/1/07583.jpg
Image : train/1/07585.jpg
Image : train/1/07587.jpg
Image : train/1/07592.jpg
Image : train/1/07594.jpg
Image : train/1/07595.jpg
Image : train/1/07596.jpg
Image : train/1/07600.jpg
Image : train/1/07601.jpg
Image : train/1/07602.jpg
Image : train/1/07603.jpg
Image : train/1/07604.jpg
Image : train/1/07605.jpg
Image : train/1/07607.jpg
Image : train/1/07608.jpg
Image : train/1/07611.jpg
Image : train/1/07612.jpg
Image : train/1/07614.jpg
Image : train/1/07615.jpg
Image : train/1/07619.jpg
Image : train/1/07620.jpg
Image : train/1/07621.jpg
Image : train/1/07622.jpg
Image : train/1/07623.jpg
Image : train/1/07626.jpg
Image : train/1/07628.jpg
Image : train/1/07633.jpg
Image : train/1/07634.jpg
Image : trai

Image : train/1/08226.jpg
Image : train/1/08227.jpg
Image : train/1/08228.jpg
Image : train/1/08230.jpg
Image : train/1/08231.jpg
Image : train/1/08232.jpg
Image : train/1/08235.jpg
Image : train/1/08236.jpg
Image : train/1/08240.jpg
Image : train/1/08243.jpg
Image : train/1/08247.jpg
Image : train/1/08248.jpg
Image : train/1/08249.jpg
Image : train/1/08252.jpg
Image : train/1/08255.jpg
Image : train/1/08257.jpg
Image : train/1/08260.jpg
Image : train/1/08261.jpg
Image : train/1/08263.jpg
Image : train/1/08264.jpg
Image : train/1/08265.jpg
Image : train/1/08267.jpg
Image : train/1/08268.jpg
Image : train/1/08270.jpg
Image : train/1/08271.jpg
Image : train/1/08272.jpg
Image : train/1/08273.jpg
Image : train/1/08277.jpg
Image : train/1/08278.jpg
Image : train/1/08280.jpg
Image : train/1/08282.jpg
Image : train/1/08285.jpg
Image : train/1/08287.jpg
Image : train/1/08290.jpg
Image : train/1/08295.jpg
Image : train/1/08296.jpg
Image : train/1/08298.jpg
Image : train/1/08300.jpg
Image : trai

Image : train/1/08904.jpg
Image : train/1/08905.jpg
Image : train/1/08909.jpg
Image : train/1/08910.jpg
Image : train/1/08912.jpg
Image : train/1/08913.jpg
Image : train/1/08914.jpg
Image : train/1/08916.jpg
Image : train/1/08917.jpg
Image : train/1/08918.jpg
Image : train/1/08921.jpg
Image : train/1/08923.jpg
Image : train/1/08929.jpg
Image : train/1/08931.jpg
Image : train/1/08938.jpg
Image : train/1/08943.jpg
Image : train/1/08945.jpg
Image : train/1/08947.jpg
Image : train/1/08949.jpg
Image : train/1/08950.jpg
Image : train/1/08951.jpg
Image : train/1/08952.jpg
Image : train/1/08953.jpg
Image : train/1/08957.jpg
Image : train/1/08959.jpg
Image : train/1/08960.jpg
Image : train/1/08961.jpg
Image : train/1/08962.jpg
Image : train/1/08964.jpg
Image : train/1/08965.jpg
Image : train/1/08966.jpg
Image : train/1/08969.jpg
Image : train/1/08970.jpg
Image : train/1/08971.jpg
Image : train/1/08974.jpg
Image : train/1/08975.jpg
Image : train/1/08978.jpg
Image : train/1/08980.jpg
Image : trai

Image : train/1/09562.jpg
Image : train/1/09563.jpg
Image : train/1/09564.jpg
Image : train/1/09566.jpg
Image : train/1/09568.jpg
Image : train/1/09569.jpg
Image : train/1/09572.jpg
Image : train/1/09575.jpg
Image : train/1/09577.jpg
Image : train/1/09579.jpg
Image : train/1/09580.jpg
Image : train/1/09582.jpg
Image : train/1/09584.jpg
Image : train/1/09587.jpg
Image : train/1/09589.jpg
Image : train/1/09592.jpg
Image : train/1/09593.jpg
Image : train/1/09597.jpg
Image : train/1/09598.jpg
Image : train/1/09600.jpg
Image : train/1/09601.jpg
Image : train/1/09602.jpg
Image : train/1/09604.jpg
Image : train/1/09605.jpg
Image : train/1/09607.jpg
Image : train/1/09609.jpg
Image : train/1/09612.jpg
Image : train/1/09619.jpg
Image : train/1/09620.jpg
Image : train/1/09621.jpg
Image : train/1/09623.jpg
Image : train/1/09624.jpg
Image : train/1/09626.jpg
Image : train/1/09628.jpg
Image : train/1/09629.jpg
Image : train/1/09630.jpg
Image : train/1/09631.jpg
Image : train/1/09634.jpg
Image : trai

Image : train/1/10246.jpg
Image : train/1/10247.jpg
Image : train/1/10248.jpg
Image : train/1/10249.jpg
Image : train/1/10252.jpg
Image : train/1/10256.jpg
Image : train/1/10258.jpg
Image : train/1/10260.jpg
Image : train/1/10262.jpg
Image : train/1/10265.jpg
Image : train/1/10270.jpg
Image : train/1/10271.jpg
Image : train/1/10273.jpg
Image : train/1/10274.jpg
Image : train/1/10276.jpg
Image : train/1/10277.jpg
Image : train/1/10280.jpg
Image : train/1/10281.jpg
Image : train/1/10284.jpg
Image : train/1/10286.jpg
Image : train/1/10287.jpg
Image : train/1/10290.jpg
Image : train/1/10291.jpg
Image : train/1/10294.jpg
Image : train/1/10297.jpg
Image : train/1/10302.jpg
Image : train/1/10303.jpg
Image : train/1/10304.jpg
Image : train/1/10306.jpg
Image : train/1/10307.jpg
Image : train/1/10309.jpg
Image : train/1/10310.jpg
Image : train/1/10312.jpg
Image : train/1/10314.jpg
Image : train/1/10315.jpg
Image : train/1/10319.jpg
Image : train/1/10322.jpg
Image : train/1/10323.jpg
Image : trai

Image : train/1/10877.jpg
Image : train/1/10879.jpg
Image : train/1/10882.jpg
Image : train/1/10883.jpg
Image : train/1/10885.jpg
Image : train/1/10887.jpg
Image : train/1/10888.jpg
Image : train/1/10890.jpg
Image : train/1/10891.jpg
Image : train/1/10896.jpg
Image : train/1/10897.jpg
Image : train/1/10901.jpg
Image : train/1/10902.jpg
Image : train/1/10903.jpg
Image : train/1/10904.jpg
Image : train/1/10905.jpg
Image : train/1/10906.jpg
Image : train/1/10908.jpg
Image : train/1/10910.jpg
Image : train/1/10911.jpg
Image : train/1/10913.jpg
Image : train/1/10920.jpg
Image : train/1/10922.jpg
Image : train/1/10924.jpg
Image : train/1/10925.jpg
Image : train/1/10929.jpg
Image : train/1/10930.jpg
Image : train/1/10932.jpg
Image : train/1/10934.jpg
Image : train/1/10935.jpg
Image : train/1/10937.jpg
Image : train/1/10940.jpg
Image : train/1/10943.jpg
Image : train/1/10945.jpg
Image : train/1/10946.jpg
Image : train/1/10949.jpg
Image : train/1/10950.jpg
Image : train/1/10951.jpg
Image : trai

Image : train/1/11569.jpg
Image : train/1/11571.jpg
Image : train/1/11573.jpg
Image : train/1/11575.jpg
Image : train/1/11576.jpg
Image : train/1/11579.jpg
Image : train/1/11583.jpg
Image : train/1/11585.jpg
Image : train/1/11586.jpg
Image : train/1/11587.jpg
Image : train/1/11590.jpg
Image : train/1/11591.jpg
Image : train/1/11592.jpg
Image : train/1/11595.jpg
Image : train/1/11596.jpg
Image : train/1/11597.jpg
Image : train/1/11598.jpg
Image : train/1/11599.jpg
Image : train/1/11600.jpg
Image : train/1/11601.jpg
Image : train/1/11603.jpg
Image : train/1/11607.jpg
Image : train/1/11611.jpg
Image : train/1/11612.jpg
Image : train/1/11614.jpg
Image : train/1/11615.jpg
Image : train/1/11619.jpg
Image : train/1/11620.jpg
Image : train/1/11623.jpg
Image : train/1/11625.jpg
Image : train/1/11626.jpg
Image : train/1/11627.jpg
Image : train/1/11629.jpg
Image : train/1/11637.jpg
Image : train/1/11638.jpg
Image : train/1/11639.jpg
Image : train/1/11644.jpg
Image : train/1/11645.jpg
Image : trai

Image : train/1/12201.jpg
Image : train/1/12202.jpg
Image : train/1/12204.jpg
Image : train/1/12207.jpg
Image : train/1/12209.jpg
Image : train/1/12210.jpg
Image : train/1/12212.jpg
Image : train/1/12216.jpg
Image : train/1/12218.jpg
Image : train/1/12220.jpg
Image : train/1/12221.jpg
Image : train/1/12224.jpg
Image : train/1/12229.jpg
Image : train/1/12230.jpg
Image : train/1/12231.jpg
Image : train/1/12232.jpg
Image : train/1/12234.jpg
Image : train/1/12235.jpg
Image : train/1/12236.jpg
Image : train/1/12237.jpg
Image : train/1/12238.jpg
Image : train/1/12239.jpg
Image : train/1/12240.jpg
Image : train/1/12242.jpg
Image : train/1/12251.jpg
Image : train/1/12252.jpg
Image : train/1/12253.jpg
Image : train/1/12259.jpg
Image : train/1/12260.jpg
Image : train/1/12261.jpg
Image : train/1/12262.jpg
Image : train/1/12263.jpg
Image : train/1/12264.jpg
Image : train/1/12268.jpg
Image : train/1/12269.jpg
Image : train/1/12270.jpg
Image : train/1/12271.jpg
Image : train/1/12273.jpg
Image : trai

Image : train/1/12833.jpg
Image : train/1/12834.jpg
Image : train/1/12838.jpg
Image : train/1/12841.jpg
Image : train/1/12842.jpg
Image : train/1/12846.jpg
Image : train/1/12847.jpg
Image : train/1/12855.jpg
Image : train/1/12856.jpg
Image : train/1/12859.jpg
Image : train/1/12861.jpg
Image : train/1/12864.jpg
Image : train/1/12867.jpg
Image : train/1/12876.jpg
Image : train/1/12877.jpg
Image : train/1/12886.jpg
Image : train/1/12888.jpg
Image : train/1/12890.jpg
Image : train/1/12892.jpg
Image : train/1/12895.jpg
Image : train/1/12896.jpg
Image : train/1/12899.jpg
Image : train/1/12900.jpg
Image : train/1/12904.jpg
Image : train/1/12907.jpg
Image : train/1/12910.jpg
Image : train/1/12911.jpg
Image : train/1/12912.jpg
Image : train/1/12919.jpg
Image : train/1/12923.jpg
Image : train/1/12924.jpg
Image : train/1/12925.jpg
Image : train/1/12927.jpg
Image : train/1/12928.jpg
Image : train/1/12932.jpg
Image : train/1/12934.jpg
Image : train/1/12938.jpg
Image : train/1/12941.jpg
Image : trai

Image : train/1/13518.jpg
Image : train/1/13521.jpg
Image : train/1/13526.jpg
Image : train/1/13530.jpg
Image : train/1/13531.jpg
Image : train/1/13534.jpg
Image : train/1/13536.jpg
Image : train/1/13538.jpg
Image : train/1/13542.jpg
Image : train/1/13543.jpg
Image : train/1/13544.jpg
Image : train/1/13545.jpg
Image : train/1/13548.jpg
Image : train/1/13559.jpg
Image : train/1/13560.jpg
Image : train/1/13561.jpg
Image : train/1/13562.jpg
Image : train/1/13564.jpg
Image : train/1/13565.jpg
Image : train/1/13568.jpg
Image : train/1/13571.jpg
Image : train/1/13572.jpg
Image : train/1/13574.jpg
Image : train/1/13575.jpg
Image : train/1/13576.jpg
Image : train/1/13579.jpg
Image : train/1/13582.jpg
Image : train/1/13585.jpg
Image : train/1/13587.jpg
Image : train/1/13590.jpg
Image : train/1/13592.jpg
Image : train/1/13593.jpg
Image : train/1/13596.jpg
Image : train/1/13597.jpg
Image : train/1/13598.jpg
Image : train/1/13599.jpg
Image : train/1/13601.jpg
Image : train/1/13603.jpg
Image : trai

Image : train/1/14145.jpg
Image : train/1/14149.jpg
Image : train/1/14150.jpg
Image : train/1/14151.jpg
Image : train/1/14156.jpg
Image : train/1/14158.jpg
Image : train/1/14160.jpg
Image : train/1/14164.jpg
Image : train/1/14166.jpg
Image : train/1/14167.jpg
Image : train/1/14169.jpg
Image : train/1/14173.jpg
Image : train/1/14174.jpg
Image : train/1/14184.jpg
Image : train/1/14185.jpg
Image : train/1/14186.jpg
Image : train/1/14188.jpg
Image : train/1/14189.jpg
Image : train/1/14190.jpg
Image : train/1/14191.jpg
Image : train/1/14192.jpg
Image : train/1/14193.jpg
Image : train/1/14194.jpg
Image : train/1/14195.jpg
Image : train/1/14197.jpg
Image : train/1/14203.jpg
Image : train/1/14204.jpg
Image : train/1/14211.jpg
Image : train/1/14212.jpg
Image : train/1/14213.jpg
Image : train/1/14218.jpg
Image : train/1/14219.jpg
Image : train/1/14220.jpg
Image : train/1/14221.jpg
Image : train/1/14223.jpg
Image : train/1/14225.jpg
Image : train/1/14227.jpg
Image : train/1/14228.jpg
Image : trai

Image : train/1/14788.jpg
Image : train/1/14790.jpg
Image : train/1/14792.jpg
Image : train/1/14793.jpg
Image : train/1/14794.jpg
Image : train/1/14797.jpg
Image : train/1/14798.jpg
Image : train/1/14799.jpg
Image : train/1/14803.jpg
Image : train/1/14804.jpg
Image : train/1/14805.jpg
Image : train/1/14806.jpg
Image : train/1/14807.jpg
Image : train/1/14808.jpg
Image : train/1/14814.jpg
Image : train/1/14818.jpg
Image : train/1/14820.jpg
Image : train/1/14822.jpg
Image : train/1/14823.jpg
Image : train/1/14827.jpg
Image : train/1/14828.jpg
Image : train/1/14830.jpg
Image : train/1/14831.jpg
Image : train/1/14832.jpg
Image : train/1/14834.jpg
Image : train/1/14837.jpg
Image : train/1/14838.jpg
Image : train/1/14839.jpg
Image : train/1/14842.jpg
Image : train/1/14843.jpg
Image : train/1/14844.jpg
Image : train/1/14848.jpg
Image : train/1/14851.jpg
Image : train/1/14853.jpg
Image : train/1/14855.jpg
Image : train/1/14856.jpg
Image : train/1/14857.jpg
Image : train/1/14858.jpg
Image : trai

Image : train/1/15451.jpg
Image : train/1/15452.jpg
Image : train/1/15458.jpg
Image : train/1/15459.jpg
Image : train/1/15463.jpg
Image : train/1/15464.jpg
Image : train/1/15468.jpg
Image : train/1/15469.jpg
Image : train/1/15470.jpg
Image : train/1/15473.jpg
Image : train/1/15474.jpg
Image : train/1/15477.jpg
Image : train/1/15478.jpg
Image : train/1/15486.jpg
Image : train/1/15487.jpg
Image : train/1/15488.jpg
Image : train/1/15490.jpg
Image : train/1/15492.jpg
Image : train/1/15494.jpg
Image : train/1/15497.jpg
Image : train/1/15500.jpg
Image : train/1/15502.jpg
Image : train/1/15504.jpg
Image : train/1/15508.jpg
Image : train/1/15511.jpg
Image : train/1/15512.jpg
Image : train/1/15514.jpg
Image : train/1/15515.jpg
Image : train/1/15518.jpg
Image : train/1/15519.jpg
Image : train/1/15520.jpg
Image : train/1/15522.jpg
Image : train/1/15524.jpg
Image : train/1/15525.jpg
Image : train/1/15526.jpg
Image : train/1/15528.jpg
Image : train/1/15530.jpg
Image : train/1/15531.jpg
Image : trai

Image : train/1/16111.jpg
Image : train/1/16112.jpg
Image : train/1/16115.jpg
Image : train/1/16117.jpg
Image : train/1/16118.jpg
Image : train/1/16119.jpg
Image : train/1/16121.jpg
Image : train/1/16122.jpg
Image : train/1/16125.jpg
Image : train/1/16126.jpg
Image : train/1/16127.jpg
Image : train/1/16128.jpg
Image : train/1/16130.jpg
Image : train/1/16136.jpg
Image : train/1/16139.jpg
Image : train/1/16140.jpg
Image : train/1/16141.jpg
Image : train/1/16142.jpg
Image : train/1/16143.jpg
Image : train/1/16146.jpg
Image : train/1/16147.jpg
Image : train/1/16153.jpg
Image : train/1/16154.jpg
Image : train/1/16155.jpg
Image : train/1/16157.jpg
Image : train/1/16158.jpg
Image : train/1/16159.jpg
Image : train/1/16160.jpg
Image : train/1/16161.jpg
Image : train/1/16165.jpg
Image : train/1/16166.jpg
Image : train/1/16168.jpg
Image : train/1/16169.jpg
Image : train/1/16173.jpg
Image : train/1/16174.jpg
Image : train/1/16175.jpg
Image : train/1/16176.jpg
Image : train/1/16179.jpg
Image : trai

Image : train/1/16753.jpg
Image : train/1/16754.jpg
Image : train/1/16755.jpg
Image : train/1/16757.jpg
Image : train/1/16763.jpg
Image : train/1/16767.jpg
Image : train/1/16769.jpg
Image : train/1/16771.jpg
Image : train/1/16772.jpg
Image : train/1/16773.jpg
Image : train/1/16774.jpg
Image : train/1/16775.jpg
Image : train/1/16777.jpg
Image : train/1/16782.jpg
Image : train/1/16786.jpg
Image : train/1/16789.jpg
Image : train/1/16790.jpg
Image : train/1/16794.jpg
Image : train/1/16795.jpg
Image : train/1/16798.jpg
Image : train/1/16800.jpg
Image : train/1/16803.jpg
Image : train/1/16807.jpg
Image : train/1/16810.jpg
Image : train/1/16811.jpg
Image : train/1/16812.jpg
Image : train/1/16813.jpg
Image : train/1/16814.jpg
Image : train/1/16815.jpg
Image : train/1/16816.jpg
Image : train/1/16817.jpg
Image : train/1/16819.jpg
Image : train/1/16822.jpg
Image : train/1/16823.jpg
Image : train/1/16824.jpg
Image : train/1/16825.jpg
Image : train/1/16826.jpg
Image : train/1/16829.jpg
Image : trai

Image : train/1/17404.jpg
Image : train/1/17405.jpg
Image : train/1/17409.jpg
Image : train/1/17410.jpg
Image : train/1/17413.jpg
Image : train/1/17414.jpg
Image : train/1/17415.jpg
Image : train/1/17416.jpg
Image : train/1/17418.jpg
Image : train/1/17419.jpg
Image : train/1/17420.jpg
Image : train/1/17422.jpg
Image : train/1/17424.jpg
Image : train/1/17427.jpg
Image : train/1/17429.jpg
Image : train/1/17430.jpg
Image : train/1/17432.jpg
Image : train/1/17435.jpg
Image : train/1/17436.jpg
Image : train/1/17439.jpg
Image : train/1/17440.jpg
Image : train/1/17441.jpg
Image : train/1/17442.jpg
Image : train/1/17445.jpg
Image : train/1/17446.jpg
Image : train/1/17447.jpg
Image : train/1/17449.jpg
Image : train/1/17452.jpg
Image : train/1/17454.jpg
Image : train/1/17457.jpg
Image : train/1/17459.jpg
Image : train/1/17461.jpg
Image : train/1/17464.jpg
Image : train/1/17465.jpg
Image : train/1/17466.jpg
Image : train/1/17467.jpg
Image : train/1/17468.jpg
Image : train/1/17469.jpg
Image : trai

Image : train/1/18098.jpg
Image : train/1/18101.jpg
Image : train/1/18102.jpg
Image : train/1/18104.jpg
Image : train/1/18105.jpg
Image : train/1/18109.jpg
Image : train/1/18110.jpg
Image : train/1/18113.jpg
Image : train/1/18117.jpg
Image : train/1/18118.jpg
Image : train/1/18119.jpg
Image : train/1/18123.jpg
Image : train/1/18126.jpg
Image : train/1/18127.jpg
Image : train/1/18130.jpg
Image : train/1/18132.jpg
Image : train/1/18133.jpg
Image : train/1/18134.jpg
Image : train/1/18135.jpg
Image : train/1/18140.jpg
Image : train/1/18143.jpg
Image : train/1/18148.jpg
Image : train/1/18149.jpg
Image : train/1/18150.jpg
Image : train/1/18151.jpg
Image : train/1/18152.jpg
Image : train/1/18155.jpg
Image : train/1/18158.jpg
Image : train/1/18159.jpg
Image : train/1/18160.jpg
Image : train/1/18163.jpg
Image : train/1/18164.jpg
Image : train/1/18166.jpg
Image : train/1/18167.jpg
Image : train/1/18169.jpg
Image : train/1/18170.jpg
Image : train/1/18172.jpg
Image : train/1/18173.jpg
Image : trai

Image : train/1/18791.jpg
Image : train/1/18793.jpg
Image : train/1/18795.jpg
Image : train/1/18796.jpg
Image : train/1/18797.jpg
Image : train/1/18798.jpg
Image : train/1/18799.jpg
Image : train/1/18800.jpg
Image : train/1/18801.jpg
Image : train/1/18802.jpg
Image : train/1/18803.jpg
Image : train/1/18804.jpg
Image : train/1/18805.jpg
Image : train/1/18806.jpg
Image : train/1/18809.jpg
Image : train/1/18811.jpg
Image : train/1/18812.jpg
Image : train/1/18814.jpg
Image : train/1/18815.jpg
Image : train/1/18818.jpg
Image : train/1/18819.jpg
Image : train/1/18822.jpg
Image : train/1/18825.jpg
Image : train/1/18826.jpg
Image : train/1/18827.jpg
Image : train/1/18828.jpg
Image : train/1/18829.jpg
Image : train/1/18830.jpg
Image : train/1/18831.jpg
Image : train/1/18832.jpg
Image : train/1/18833.jpg
Image : train/1/18835.jpg
Image : train/1/18840.jpg
Image : train/1/18841.jpg
Image : train/1/18846.jpg
Image : train/1/18847.jpg
Image : train/1/18848.jpg
Image : train/1/18849.jpg
Image : trai

Image : train/1/19419.jpg
Image : train/1/19420.jpg
Image : train/1/19421.jpg
Image : train/1/19423.jpg
Image : train/1/19426.jpg
Image : train/1/19428.jpg
Image : train/1/19431.jpg
Image : train/1/19435.jpg
Image : train/1/19438.jpg
Image : train/1/19439.jpg
Image : train/1/19440.jpg
Image : train/1/19441.jpg
Image : train/1/19442.jpg
Image : train/1/19447.jpg
Image : train/1/19449.jpg
Image : train/1/19451.jpg
Image : train/1/19452.jpg
Image : train/1/19453.jpg
Image : train/1/19456.jpg
Image : train/1/19457.jpg
Image : train/1/19458.jpg
Image : train/1/19459.jpg
Image : train/1/19462.jpg
Image : train/1/19463.jpg
Image : train/1/19464.jpg
Image : train/1/19470.jpg
Image : train/1/19471.jpg
Image : train/1/19472.jpg
Image : train/1/19475.jpg
Image : train/1/19476.jpg
Image : train/1/19477.jpg
Image : train/1/19480.jpg
Image : train/1/19481.jpg
Image : train/1/19487.jpg
Image : train/1/19489.jpg
Image : train/1/19490.jpg
Image : train/1/19491.jpg
Image : train/1/19492.jpg
Image : trai

Image : train/1/20133.jpg
Image : train/1/20134.jpg
Image : train/1/20137.jpg
Image : train/1/20143.jpg
Image : train/1/20144.jpg
Image : train/1/20145.jpg
Image : train/1/20149.jpg
Image : train/1/20153.jpg
Image : train/1/20155.jpg
Image : train/1/20157.jpg
Image : train/1/20159.jpg
Image : train/1/20161.jpg
Image : train/1/20165.jpg
Image : train/1/20166.jpg
Image : train/1/20168.jpg
Image : train/1/20169.jpg
Image : train/1/20172.jpg
Image : train/1/20175.jpg
Image : train/1/20177.jpg
Image : train/1/20180.jpg
Image : train/1/20182.jpg
Image : train/1/20183.jpg
Image : train/1/20188.jpg
Image : train/1/20192.jpg
Image : train/1/20196.jpg
Image : train/1/20197.jpg
Image : train/1/20199.jpg
Image : train/1/20201.jpg
Image : train/1/20202.jpg
Image : train/1/20204.jpg
Image : train/1/20205.jpg
Image : train/1/20206.jpg
Image : train/1/20211.jpg
Image : train/1/20212.jpg
Image : train/1/20213.jpg
Image : train/1/20216.jpg
Image : train/1/20217.jpg
Image : train/1/20218.jpg
Image : trai

Image : train/1/20832.jpg
Image : train/1/20836.jpg
Image : train/1/20837.jpg
Image : train/1/20841.jpg
Image : train/1/20842.jpg
Image : train/1/20845.jpg
Image : train/1/20846.jpg
Image : train/1/20847.jpg
Image : train/1/20848.jpg
Image : train/1/20849.jpg
Image : train/1/20850.jpg
Image : train/1/20852.jpg
Image : train/1/20853.jpg
Image : train/1/20854.jpg
Image : train/1/20857.jpg
Image : train/1/20859.jpg
Image : train/1/20862.jpg
Image : train/1/20863.jpg
Image : train/1/20865.jpg
Image : train/1/20866.jpg
Image : train/1/20868.jpg
Image : train/1/20869.jpg
Image : train/1/20871.jpg
Image : train/1/20874.jpg
Image : train/1/20875.jpg
Image : train/1/20876.jpg
Image : train/1/20877.jpg
Image : train/1/20878.jpg
Image : train/1/20879.jpg
Image : train/1/20881.jpg
Image : train/1/20883.jpg
Image : train/1/20884.jpg
Image : train/1/20888.jpg
Image : train/1/20890.jpg
Image : train/1/20892.jpg
Image : train/1/20893.jpg
Image : train/1/20894.jpg
Image : train/1/20895.jpg
Image : trai

Image : train/1/21448.jpg
Image : train/1/21449.jpg
Image : train/1/21450.jpg
Image : train/1/21452.jpg
Image : train/1/21454.jpg
Image : train/1/21455.jpg
Image : train/1/21456.jpg
Image : train/1/21459.jpg
Image : train/1/21462.jpg
Image : train/1/21463.jpg
Image : train/1/21465.jpg
Image : train/1/21466.jpg
Image : train/1/21470.jpg
Image : train/1/21472.jpg
Image : train/1/21473.jpg
Image : train/1/21475.jpg
Image : train/1/21476.jpg
Image : train/1/21478.jpg
Image : train/1/21480.jpg
Image : train/1/21481.jpg
Image : train/1/21482.jpg
Image : train/1/21484.jpg
Image : train/1/21485.jpg
Image : train/1/21486.jpg
Image : train/1/21489.jpg
Image : train/1/21490.jpg
Image : train/1/21491.jpg
Image : train/1/21492.jpg
Image : train/1/21494.jpg
Image : train/1/21495.jpg
Image : train/1/21497.jpg
Image : train/1/21499.jpg
Image : train/1/21500.jpg
Image : train/1/21503.jpg
Image : train/1/21506.jpg
Image : train/1/21508.jpg
Image : train/1/21509.jpg
Image : train/1/21510.jpg
Image : trai

UN-INFECTED DATA

In [5]:
labels = os.listdir(uninfected_dir)

for label in labels:
    
    filename = uninfected_dir +'/'+label
    rot_img = image.load_img(filename, target_size=(128,128,3), color_mode="rgb")

    img = image.img_to_array(rot_img)
    img = img/255    
    D.append((img, 0))

    print("Image : "+ filename)

Image : train/0/00003.jpg
Image : train/0/00004.jpg
Image : train/0/00006.jpg
Image : train/0/00009.jpg
Image : train/0/00017.jpg
Image : train/0/00018.jpg
Image : train/0/00019.jpg
Image : train/0/00021.jpg
Image : train/0/00022.jpg
Image : train/0/00023.jpg
Image : train/0/00024.jpg
Image : train/0/00028.jpg
Image : train/0/00029.jpg
Image : train/0/00030.jpg
Image : train/0/00032.jpg
Image : train/0/00033.jpg
Image : train/0/00038.jpg
Image : train/0/00042.jpg
Image : train/0/00044.jpg
Image : train/0/00045.jpg
Image : train/0/00047.jpg
Image : train/0/00048.jpg
Image : train/0/00049.jpg
Image : train/0/00050.jpg
Image : train/0/00052.jpg
Image : train/0/00053.jpg
Image : train/0/00055.jpg
Image : train/0/00057.jpg
Image : train/0/00058.jpg
Image : train/0/00060.jpg
Image : train/0/00063.jpg
Image : train/0/00065.jpg
Image : train/0/00066.jpg
Image : train/0/00067.jpg
Image : train/0/00069.jpg
Image : train/0/00073.jpg
Image : train/0/00074.jpg
Image : train/0/00075.jpg
Image : trai

Image : train/0/00600.jpg
Image : train/0/00604.jpg
Image : train/0/00605.jpg
Image : train/0/00606.jpg
Image : train/0/00607.jpg
Image : train/0/00610.jpg
Image : train/0/00611.jpg
Image : train/0/00615.jpg
Image : train/0/00617.jpg
Image : train/0/00618.jpg
Image : train/0/00619.jpg
Image : train/0/00621.jpg
Image : train/0/00622.jpg
Image : train/0/00624.jpg
Image : train/0/00625.jpg
Image : train/0/00627.jpg
Image : train/0/00628.jpg
Image : train/0/00629.jpg
Image : train/0/00630.jpg
Image : train/0/00635.jpg
Image : train/0/00636.jpg
Image : train/0/00637.jpg
Image : train/0/00638.jpg
Image : train/0/00640.jpg
Image : train/0/00643.jpg
Image : train/0/00644.jpg
Image : train/0/00645.jpg
Image : train/0/00646.jpg
Image : train/0/00649.jpg
Image : train/0/00650.jpg
Image : train/0/00651.jpg
Image : train/0/00652.jpg
Image : train/0/00653.jpg
Image : train/0/00655.jpg
Image : train/0/00656.jpg
Image : train/0/00659.jpg
Image : train/0/00662.jpg
Image : train/0/00665.jpg
Image : trai

Image : train/0/01202.jpg
Image : train/0/01203.jpg
Image : train/0/01204.jpg
Image : train/0/01206.jpg
Image : train/0/01207.jpg
Image : train/0/01210.jpg
Image : train/0/01211.jpg
Image : train/0/01212.jpg
Image : train/0/01214.jpg
Image : train/0/01217.jpg
Image : train/0/01218.jpg
Image : train/0/01219.jpg
Image : train/0/01220.jpg
Image : train/0/01221.jpg
Image : train/0/01223.jpg
Image : train/0/01226.jpg
Image : train/0/01228.jpg
Image : train/0/01231.jpg
Image : train/0/01232.jpg
Image : train/0/01234.jpg
Image : train/0/01235.jpg
Image : train/0/01236.jpg
Image : train/0/01237.jpg
Image : train/0/01243.jpg
Image : train/0/01244.jpg
Image : train/0/01246.jpg
Image : train/0/01247.jpg
Image : train/0/01249.jpg
Image : train/0/01250.jpg
Image : train/0/01251.jpg
Image : train/0/01252.jpg
Image : train/0/01256.jpg
Image : train/0/01258.jpg
Image : train/0/01259.jpg
Image : train/0/01260.jpg
Image : train/0/01261.jpg
Image : train/0/01262.jpg
Image : train/0/01263.jpg
Image : trai

Image : train/0/01824.jpg
Image : train/0/01828.jpg
Image : train/0/01829.jpg
Image : train/0/01830.jpg
Image : train/0/01835.jpg
Image : train/0/01836.jpg
Image : train/0/01840.jpg
Image : train/0/01841.jpg
Image : train/0/01842.jpg
Image : train/0/01844.jpg
Image : train/0/01846.jpg
Image : train/0/01847.jpg
Image : train/0/01848.jpg
Image : train/0/01852.jpg
Image : train/0/01853.jpg
Image : train/0/01857.jpg
Image : train/0/01858.jpg
Image : train/0/01859.jpg
Image : train/0/01860.jpg
Image : train/0/01861.jpg
Image : train/0/01862.jpg
Image : train/0/01865.jpg
Image : train/0/01866.jpg
Image : train/0/01867.jpg
Image : train/0/01869.jpg
Image : train/0/01874.jpg
Image : train/0/01878.jpg
Image : train/0/01879.jpg
Image : train/0/01880.jpg
Image : train/0/01881.jpg
Image : train/0/01882.jpg
Image : train/0/01884.jpg
Image : train/0/01886.jpg
Image : train/0/01887.jpg
Image : train/0/01888.jpg
Image : train/0/01889.jpg
Image : train/0/01890.jpg
Image : train/0/01891.jpg
Image : trai

Image : train/0/02486.jpg
Image : train/0/02488.jpg
Image : train/0/02490.jpg
Image : train/0/02491.jpg
Image : train/0/02493.jpg
Image : train/0/02494.jpg
Image : train/0/02499.jpg
Image : train/0/02500.jpg
Image : train/0/02501.jpg
Image : train/0/02502.jpg
Image : train/0/02504.jpg
Image : train/0/02505.jpg
Image : train/0/02506.jpg
Image : train/0/02508.jpg
Image : train/0/02510.jpg
Image : train/0/02511.jpg
Image : train/0/02512.jpg
Image : train/0/02515.jpg
Image : train/0/02519.jpg
Image : train/0/02520.jpg
Image : train/0/02522.jpg
Image : train/0/02523.jpg
Image : train/0/02527.jpg
Image : train/0/02528.jpg
Image : train/0/02529.jpg
Image : train/0/02532.jpg
Image : train/0/02536.jpg
Image : train/0/02539.jpg
Image : train/0/02540.jpg
Image : train/0/02541.jpg
Image : train/0/02543.jpg
Image : train/0/02544.jpg
Image : train/0/02546.jpg
Image : train/0/02547.jpg
Image : train/0/02549.jpg
Image : train/0/02551.jpg
Image : train/0/02554.jpg
Image : train/0/02555.jpg
Image : trai

Image : train/0/03184.jpg
Image : train/0/03187.jpg
Image : train/0/03188.jpg
Image : train/0/03189.jpg
Image : train/0/03190.jpg
Image : train/0/03191.jpg
Image : train/0/03192.jpg
Image : train/0/03193.jpg
Image : train/0/03197.jpg
Image : train/0/03198.jpg
Image : train/0/03199.jpg
Image : train/0/03203.jpg
Image : train/0/03205.jpg
Image : train/0/03207.jpg
Image : train/0/03209.jpg
Image : train/0/03212.jpg
Image : train/0/03214.jpg
Image : train/0/03216.jpg
Image : train/0/03217.jpg
Image : train/0/03221.jpg
Image : train/0/03222.jpg
Image : train/0/03224.jpg
Image : train/0/03225.jpg
Image : train/0/03226.jpg
Image : train/0/03227.jpg
Image : train/0/03228.jpg
Image : train/0/03229.jpg
Image : train/0/03230.jpg
Image : train/0/03236.jpg
Image : train/0/03239.jpg
Image : train/0/03240.jpg
Image : train/0/03241.jpg
Image : train/0/03244.jpg
Image : train/0/03245.jpg
Image : train/0/03249.jpg
Image : train/0/03253.jpg
Image : train/0/03256.jpg
Image : train/0/03257.jpg
Image : trai

Image : train/0/03882.jpg
Image : train/0/03883.jpg
Image : train/0/03884.jpg
Image : train/0/03885.jpg
Image : train/0/03887.jpg
Image : train/0/03889.jpg
Image : train/0/03893.jpg
Image : train/0/03894.jpg
Image : train/0/03895.jpg
Image : train/0/03896.jpg
Image : train/0/03900.jpg
Image : train/0/03901.jpg
Image : train/0/03902.jpg
Image : train/0/03903.jpg
Image : train/0/03904.jpg
Image : train/0/03905.jpg
Image : train/0/03908.jpg
Image : train/0/03914.jpg
Image : train/0/03915.jpg
Image : train/0/03916.jpg
Image : train/0/03925.jpg
Image : train/0/03928.jpg
Image : train/0/03929.jpg
Image : train/0/03931.jpg
Image : train/0/03932.jpg
Image : train/0/03933.jpg
Image : train/0/03938.jpg
Image : train/0/03939.jpg
Image : train/0/03940.jpg
Image : train/0/03943.jpg
Image : train/0/03944.jpg
Image : train/0/03945.jpg
Image : train/0/03946.jpg
Image : train/0/03947.jpg
Image : train/0/03948.jpg
Image : train/0/03951.jpg
Image : train/0/03954.jpg
Image : train/0/03955.jpg
Image : trai

Image : train/0/04544.jpg
Image : train/0/04545.jpg
Image : train/0/04548.jpg
Image : train/0/04549.jpg
Image : train/0/04553.jpg
Image : train/0/04555.jpg
Image : train/0/04556.jpg
Image : train/0/04559.jpg
Image : train/0/04560.jpg
Image : train/0/04561.jpg
Image : train/0/04565.jpg
Image : train/0/04566.jpg
Image : train/0/04567.jpg
Image : train/0/04568.jpg
Image : train/0/04569.jpg
Image : train/0/04570.jpg
Image : train/0/04571.jpg
Image : train/0/04572.jpg
Image : train/0/04574.jpg
Image : train/0/04577.jpg
Image : train/0/04578.jpg
Image : train/0/04579.jpg
Image : train/0/04580.jpg
Image : train/0/04581.jpg
Image : train/0/04583.jpg
Image : train/0/04585.jpg
Image : train/0/04589.jpg
Image : train/0/04590.jpg
Image : train/0/04594.jpg
Image : train/0/04595.jpg
Image : train/0/04596.jpg
Image : train/0/04600.jpg
Image : train/0/04602.jpg
Image : train/0/04606.jpg
Image : train/0/04607.jpg
Image : train/0/04609.jpg
Image : train/0/04610.jpg
Image : train/0/04611.jpg
Image : trai

Image : train/0/05210.jpg
Image : train/0/05212.jpg
Image : train/0/05214.jpg
Image : train/0/05215.jpg
Image : train/0/05217.jpg
Image : train/0/05218.jpg
Image : train/0/05219.jpg
Image : train/0/05223.jpg
Image : train/0/05224.jpg
Image : train/0/05225.jpg
Image : train/0/05226.jpg
Image : train/0/05227.jpg
Image : train/0/05228.jpg
Image : train/0/05230.jpg
Image : train/0/05233.jpg
Image : train/0/05234.jpg
Image : train/0/05236.jpg
Image : train/0/05237.jpg
Image : train/0/05238.jpg
Image : train/0/05240.jpg
Image : train/0/05241.jpg
Image : train/0/05242.jpg
Image : train/0/05246.jpg
Image : train/0/05247.jpg
Image : train/0/05249.jpg
Image : train/0/05251.jpg
Image : train/0/05252.jpg
Image : train/0/05261.jpg
Image : train/0/05263.jpg
Image : train/0/05270.jpg
Image : train/0/05271.jpg
Image : train/0/05277.jpg
Image : train/0/05278.jpg
Image : train/0/05280.jpg
Image : train/0/05283.jpg
Image : train/0/05285.jpg
Image : train/0/05288.jpg
Image : train/0/05290.jpg
Image : trai

Image : train/0/05933.jpg
Image : train/0/05934.jpg
Image : train/0/05935.jpg
Image : train/0/05938.jpg
Image : train/0/05939.jpg
Image : train/0/05940.jpg
Image : train/0/05946.jpg
Image : train/0/05949.jpg
Image : train/0/05950.jpg
Image : train/0/05952.jpg
Image : train/0/05954.jpg
Image : train/0/05959.jpg
Image : train/0/05960.jpg
Image : train/0/05962.jpg
Image : train/0/05963.jpg
Image : train/0/05964.jpg
Image : train/0/05968.jpg
Image : train/0/05969.jpg
Image : train/0/05970.jpg
Image : train/0/05972.jpg
Image : train/0/05973.jpg
Image : train/0/05974.jpg
Image : train/0/05976.jpg
Image : train/0/05979.jpg
Image : train/0/05980.jpg
Image : train/0/05981.jpg
Image : train/0/05983.jpg
Image : train/0/05985.jpg
Image : train/0/05987.jpg
Image : train/0/05988.jpg
Image : train/0/05989.jpg
Image : train/0/05990.jpg
Image : train/0/05991.jpg
Image : train/0/05992.jpg
Image : train/0/05993.jpg
Image : train/0/06000.jpg
Image : train/0/06002.jpg
Image : train/0/06003.jpg
Image : trai

Image : train/0/06568.jpg
Image : train/0/06571.jpg
Image : train/0/06572.jpg
Image : train/0/06574.jpg
Image : train/0/06575.jpg
Image : train/0/06581.jpg
Image : train/0/06582.jpg
Image : train/0/06585.jpg
Image : train/0/06586.jpg
Image : train/0/06587.jpg
Image : train/0/06589.jpg
Image : train/0/06592.jpg
Image : train/0/06594.jpg
Image : train/0/06596.jpg
Image : train/0/06597.jpg
Image : train/0/06600.jpg
Image : train/0/06601.jpg
Image : train/0/06603.jpg
Image : train/0/06605.jpg
Image : train/0/06606.jpg
Image : train/0/06607.jpg
Image : train/0/06609.jpg
Image : train/0/06613.jpg
Image : train/0/06614.jpg
Image : train/0/06620.jpg
Image : train/0/06622.jpg
Image : train/0/06623.jpg
Image : train/0/06624.jpg
Image : train/0/06627.jpg
Image : train/0/06630.jpg
Image : train/0/06631.jpg
Image : train/0/06632.jpg
Image : train/0/06633.jpg
Image : train/0/06634.jpg
Image : train/0/06635.jpg
Image : train/0/06639.jpg
Image : train/0/06640.jpg
Image : train/0/06641.jpg
Image : trai

Image : train/0/07251.jpg
Image : train/0/07254.jpg
Image : train/0/07256.jpg
Image : train/0/07259.jpg
Image : train/0/07261.jpg
Image : train/0/07264.jpg
Image : train/0/07269.jpg
Image : train/0/07270.jpg
Image : train/0/07271.jpg
Image : train/0/07274.jpg
Image : train/0/07276.jpg
Image : train/0/07277.jpg
Image : train/0/07280.jpg
Image : train/0/07283.jpg
Image : train/0/07286.jpg
Image : train/0/07289.jpg
Image : train/0/07294.jpg
Image : train/0/07295.jpg
Image : train/0/07296.jpg
Image : train/0/07298.jpg
Image : train/0/07301.jpg
Image : train/0/07302.jpg
Image : train/0/07303.jpg
Image : train/0/07304.jpg
Image : train/0/07305.jpg
Image : train/0/07306.jpg
Image : train/0/07307.jpg
Image : train/0/07308.jpg
Image : train/0/07310.jpg
Image : train/0/07312.jpg
Image : train/0/07313.jpg
Image : train/0/07314.jpg
Image : train/0/07317.jpg
Image : train/0/07319.jpg
Image : train/0/07321.jpg
Image : train/0/07322.jpg
Image : train/0/07326.jpg
Image : train/0/07327.jpg
Image : trai

Image : train/0/07961.jpg
Image : train/0/07965.jpg
Image : train/0/07966.jpg
Image : train/0/07967.jpg
Image : train/0/07968.jpg
Image : train/0/07969.jpg
Image : train/0/07971.jpg
Image : train/0/07973.jpg
Image : train/0/07978.jpg
Image : train/0/07979.jpg
Image : train/0/07981.jpg
Image : train/0/07982.jpg
Image : train/0/07983.jpg
Image : train/0/07986.jpg
Image : train/0/07993.jpg
Image : train/0/07995.jpg
Image : train/0/07997.jpg
Image : train/0/07998.jpg
Image : train/0/08001.jpg
Image : train/0/08006.jpg
Image : train/0/08007.jpg
Image : train/0/08008.jpg
Image : train/0/08009.jpg
Image : train/0/08014.jpg
Image : train/0/08018.jpg
Image : train/0/08019.jpg
Image : train/0/08022.jpg
Image : train/0/08024.jpg
Image : train/0/08025.jpg
Image : train/0/08026.jpg
Image : train/0/08028.jpg
Image : train/0/08030.jpg
Image : train/0/08033.jpg
Image : train/0/08036.jpg
Image : train/0/08038.jpg
Image : train/0/08039.jpg
Image : train/0/08045.jpg
Image : train/0/08048.jpg
Image : trai

Image : train/0/08647.jpg
Image : train/0/08649.jpg
Image : train/0/08650.jpg
Image : train/0/08652.jpg
Image : train/0/08655.jpg
Image : train/0/08657.jpg
Image : train/0/08660.jpg
Image : train/0/08662.jpg
Image : train/0/08665.jpg
Image : train/0/08669.jpg
Image : train/0/08671.jpg
Image : train/0/08674.jpg
Image : train/0/08675.jpg
Image : train/0/08676.jpg
Image : train/0/08679.jpg
Image : train/0/08681.jpg
Image : train/0/08683.jpg
Image : train/0/08684.jpg
Image : train/0/08685.jpg
Image : train/0/08686.jpg
Image : train/0/08687.jpg
Image : train/0/08690.jpg
Image : train/0/08691.jpg
Image : train/0/08692.jpg
Image : train/0/08693.jpg
Image : train/0/08694.jpg
Image : train/0/08695.jpg
Image : train/0/08696.jpg
Image : train/0/08698.jpg
Image : train/0/08705.jpg
Image : train/0/08707.jpg
Image : train/0/08711.jpg
Image : train/0/08712.jpg
Image : train/0/08715.jpg
Image : train/0/08719.jpg
Image : train/0/08720.jpg
Image : train/0/08722.jpg
Image : train/0/08723.jpg
Image : trai

Image : train/0/09373.jpg
Image : train/0/09375.jpg
Image : train/0/09377.jpg
Image : train/0/09378.jpg
Image : train/0/09379.jpg
Image : train/0/09383.jpg
Image : train/0/09385.jpg
Image : train/0/09386.jpg
Image : train/0/09387.jpg
Image : train/0/09389.jpg
Image : train/0/09391.jpg
Image : train/0/09392.jpg
Image : train/0/09393.jpg
Image : train/0/09395.jpg
Image : train/0/09399.jpg
Image : train/0/09401.jpg
Image : train/0/09410.jpg
Image : train/0/09411.jpg
Image : train/0/09412.jpg
Image : train/0/09413.jpg
Image : train/0/09414.jpg
Image : train/0/09415.jpg
Image : train/0/09416.jpg
Image : train/0/09420.jpg
Image : train/0/09421.jpg
Image : train/0/09422.jpg
Image : train/0/09426.jpg
Image : train/0/09427.jpg
Image : train/0/09428.jpg
Image : train/0/09434.jpg
Image : train/0/09435.jpg
Image : train/0/09436.jpg
Image : train/0/09437.jpg
Image : train/0/09438.jpg
Image : train/0/09439.jpg
Image : train/0/09443.jpg
Image : train/0/09444.jpg
Image : train/0/09446.jpg
Image : trai

Image : train/0/10070.jpg
Image : train/0/10072.jpg
Image : train/0/10074.jpg
Image : train/0/10076.jpg
Image : train/0/10080.jpg
Image : train/0/10081.jpg
Image : train/0/10082.jpg
Image : train/0/10083.jpg
Image : train/0/10084.jpg
Image : train/0/10092.jpg
Image : train/0/10094.jpg
Image : train/0/10095.jpg
Image : train/0/10096.jpg
Image : train/0/10097.jpg
Image : train/0/10098.jpg
Image : train/0/10099.jpg
Image : train/0/10101.jpg
Image : train/0/10102.jpg
Image : train/0/10109.jpg
Image : train/0/10110.jpg
Image : train/0/10111.jpg
Image : train/0/10112.jpg
Image : train/0/10113.jpg
Image : train/0/10114.jpg
Image : train/0/10117.jpg
Image : train/0/10120.jpg
Image : train/0/10121.jpg
Image : train/0/10122.jpg
Image : train/0/10123.jpg
Image : train/0/10125.jpg
Image : train/0/10126.jpg
Image : train/0/10130.jpg
Image : train/0/10132.jpg
Image : train/0/10133.jpg
Image : train/0/10134.jpg
Image : train/0/10139.jpg
Image : train/0/10140.jpg
Image : train/0/10141.jpg
Image : trai

Image : train/0/10799.jpg
Image : train/0/10803.jpg
Image : train/0/10804.jpg
Image : train/0/10806.jpg
Image : train/0/10809.jpg
Image : train/0/10810.jpg
Image : train/0/10811.jpg
Image : train/0/10812.jpg
Image : train/0/10815.jpg
Image : train/0/10818.jpg
Image : train/0/10830.jpg
Image : train/0/10831.jpg
Image : train/0/10833.jpg
Image : train/0/10834.jpg
Image : train/0/10837.jpg
Image : train/0/10838.jpg
Image : train/0/10840.jpg
Image : train/0/10841.jpg
Image : train/0/10842.jpg
Image : train/0/10848.jpg
Image : train/0/10851.jpg
Image : train/0/10852.jpg
Image : train/0/10853.jpg
Image : train/0/10854.jpg
Image : train/0/10858.jpg
Image : train/0/10859.jpg
Image : train/0/10860.jpg
Image : train/0/10862.jpg
Image : train/0/10864.jpg
Image : train/0/10865.jpg
Image : train/0/10868.jpg
Image : train/0/10878.jpg
Image : train/0/10880.jpg
Image : train/0/10881.jpg
Image : train/0/10884.jpg
Image : train/0/10886.jpg
Image : train/0/10889.jpg
Image : train/0/10892.jpg
Image : trai

Image : train/0/11472.jpg
Image : train/0/11474.jpg
Image : train/0/11475.jpg
Image : train/0/11477.jpg
Image : train/0/11479.jpg
Image : train/0/11480.jpg
Image : train/0/11481.jpg
Image : train/0/11484.jpg
Image : train/0/11488.jpg
Image : train/0/11490.jpg
Image : train/0/11492.jpg
Image : train/0/11493.jpg
Image : train/0/11494.jpg
Image : train/0/11498.jpg
Image : train/0/11503.jpg
Image : train/0/11509.jpg
Image : train/0/11515.jpg
Image : train/0/11516.jpg
Image : train/0/11518.jpg
Image : train/0/11519.jpg
Image : train/0/11520.jpg
Image : train/0/11521.jpg
Image : train/0/11523.jpg
Image : train/0/11524.jpg
Image : train/0/11525.jpg
Image : train/0/11526.jpg
Image : train/0/11528.jpg
Image : train/0/11532.jpg
Image : train/0/11533.jpg
Image : train/0/11535.jpg
Image : train/0/11539.jpg
Image : train/0/11544.jpg
Image : train/0/11545.jpg
Image : train/0/11546.jpg
Image : train/0/11547.jpg
Image : train/0/11548.jpg
Image : train/0/11552.jpg
Image : train/0/11554.jpg
Image : trai

Image : train/0/12213.jpg
Image : train/0/12214.jpg
Image : train/0/12215.jpg
Image : train/0/12217.jpg
Image : train/0/12219.jpg
Image : train/0/12222.jpg
Image : train/0/12223.jpg
Image : train/0/12225.jpg
Image : train/0/12226.jpg
Image : train/0/12227.jpg
Image : train/0/12228.jpg
Image : train/0/12233.jpg
Image : train/0/12241.jpg
Image : train/0/12243.jpg
Image : train/0/12244.jpg
Image : train/0/12245.jpg
Image : train/0/12246.jpg
Image : train/0/12247.jpg
Image : train/0/12248.jpg
Image : train/0/12249.jpg
Image : train/0/12250.jpg
Image : train/0/12254.jpg
Image : train/0/12255.jpg
Image : train/0/12256.jpg
Image : train/0/12257.jpg
Image : train/0/12258.jpg
Image : train/0/12265.jpg
Image : train/0/12266.jpg
Image : train/0/12267.jpg
Image : train/0/12272.jpg
Image : train/0/12274.jpg
Image : train/0/12275.jpg
Image : train/0/12280.jpg
Image : train/0/12283.jpg
Image : train/0/12284.jpg
Image : train/0/12285.jpg
Image : train/0/12287.jpg
Image : train/0/12290.jpg
Image : trai

Image : train/0/12914.jpg
Image : train/0/12915.jpg
Image : train/0/12916.jpg
Image : train/0/12917.jpg
Image : train/0/12918.jpg
Image : train/0/12920.jpg
Image : train/0/12921.jpg
Image : train/0/12922.jpg
Image : train/0/12926.jpg
Image : train/0/12929.jpg
Image : train/0/12930.jpg
Image : train/0/12931.jpg
Image : train/0/12933.jpg
Image : train/0/12935.jpg
Image : train/0/12936.jpg
Image : train/0/12937.jpg
Image : train/0/12939.jpg
Image : train/0/12940.jpg
Image : train/0/12942.jpg
Image : train/0/12944.jpg
Image : train/0/12947.jpg
Image : train/0/12950.jpg
Image : train/0/12951.jpg
Image : train/0/12953.jpg
Image : train/0/12954.jpg
Image : train/0/12957.jpg
Image : train/0/12961.jpg
Image : train/0/12964.jpg
Image : train/0/12969.jpg
Image : train/0/12970.jpg
Image : train/0/12971.jpg
Image : train/0/12972.jpg
Image : train/0/12974.jpg
Image : train/0/12978.jpg
Image : train/0/12979.jpg
Image : train/0/12981.jpg
Image : train/0/12982.jpg
Image : train/0/12983.jpg
Image : trai

Image : train/0/13591.jpg
Image : train/0/13594.jpg
Image : train/0/13595.jpg
Image : train/0/13600.jpg
Image : train/0/13602.jpg
Image : train/0/13604.jpg
Image : train/0/13605.jpg
Image : train/0/13606.jpg
Image : train/0/13607.jpg
Image : train/0/13609.jpg
Image : train/0/13610.jpg
Image : train/0/13611.jpg
Image : train/0/13617.jpg
Image : train/0/13618.jpg
Image : train/0/13620.jpg
Image : train/0/13621.jpg
Image : train/0/13623.jpg
Image : train/0/13628.jpg
Image : train/0/13631.jpg
Image : train/0/13632.jpg
Image : train/0/13634.jpg
Image : train/0/13635.jpg
Image : train/0/13637.jpg
Image : train/0/13639.jpg
Image : train/0/13642.jpg
Image : train/0/13645.jpg
Image : train/0/13646.jpg
Image : train/0/13647.jpg
Image : train/0/13649.jpg
Image : train/0/13650.jpg
Image : train/0/13651.jpg
Image : train/0/13652.jpg
Image : train/0/13655.jpg
Image : train/0/13656.jpg
Image : train/0/13657.jpg
Image : train/0/13658.jpg
Image : train/0/13660.jpg
Image : train/0/13661.jpg
Image : trai

Image : train/0/14336.jpg
Image : train/0/14339.jpg
Image : train/0/14344.jpg
Image : train/0/14345.jpg
Image : train/0/14346.jpg
Image : train/0/14349.jpg
Image : train/0/14350.jpg
Image : train/0/14353.jpg
Image : train/0/14356.jpg
Image : train/0/14357.jpg
Image : train/0/14359.jpg
Image : train/0/14361.jpg
Image : train/0/14364.jpg
Image : train/0/14365.jpg
Image : train/0/14367.jpg
Image : train/0/14372.jpg
Image : train/0/14373.jpg
Image : train/0/14374.jpg
Image : train/0/14376.jpg
Image : train/0/14378.jpg
Image : train/0/14379.jpg
Image : train/0/14381.jpg
Image : train/0/14383.jpg
Image : train/0/14385.jpg
Image : train/0/14388.jpg
Image : train/0/14395.jpg
Image : train/0/14396.jpg
Image : train/0/14397.jpg
Image : train/0/14401.jpg
Image : train/0/14403.jpg
Image : train/0/14405.jpg
Image : train/0/14408.jpg
Image : train/0/14409.jpg
Image : train/0/14410.jpg
Image : train/0/14413.jpg
Image : train/0/14414.jpg
Image : train/0/14417.jpg
Image : train/0/14418.jpg
Image : trai

Image : train/0/15058.jpg
Image : train/0/15060.jpg
Image : train/0/15062.jpg
Image : train/0/15066.jpg
Image : train/0/15067.jpg
Image : train/0/15068.jpg
Image : train/0/15071.jpg
Image : train/0/15072.jpg
Image : train/0/15074.jpg
Image : train/0/15082.jpg
Image : train/0/15084.jpg
Image : train/0/15085.jpg
Image : train/0/15087.jpg
Image : train/0/15088.jpg
Image : train/0/15090.jpg
Image : train/0/15091.jpg
Image : train/0/15093.jpg
Image : train/0/15094.jpg
Image : train/0/15095.jpg
Image : train/0/15096.jpg
Image : train/0/15097.jpg
Image : train/0/15100.jpg
Image : train/0/15101.jpg
Image : train/0/15102.jpg
Image : train/0/15107.jpg
Image : train/0/15113.jpg
Image : train/0/15114.jpg
Image : train/0/15115.jpg
Image : train/0/15118.jpg
Image : train/0/15121.jpg
Image : train/0/15122.jpg
Image : train/0/15123.jpg
Image : train/0/15124.jpg
Image : train/0/15125.jpg
Image : train/0/15126.jpg
Image : train/0/15129.jpg
Image : train/0/15131.jpg
Image : train/0/15136.jpg
Image : trai

Image : train/0/15742.jpg
Image : train/0/15743.jpg
Image : train/0/15745.jpg
Image : train/0/15749.jpg
Image : train/0/15750.jpg
Image : train/0/15752.jpg
Image : train/0/15753.jpg
Image : train/0/15754.jpg
Image : train/0/15755.jpg
Image : train/0/15760.jpg
Image : train/0/15761.jpg
Image : train/0/15764.jpg
Image : train/0/15765.jpg
Image : train/0/15767.jpg
Image : train/0/15768.jpg
Image : train/0/15770.jpg
Image : train/0/15776.jpg
Image : train/0/15779.jpg
Image : train/0/15780.jpg
Image : train/0/15783.jpg
Image : train/0/15784.jpg
Image : train/0/15789.jpg
Image : train/0/15792.jpg
Image : train/0/15793.jpg
Image : train/0/15794.jpg
Image : train/0/15795.jpg
Image : train/0/15798.jpg
Image : train/0/15800.jpg
Image : train/0/15801.jpg
Image : train/0/15802.jpg
Image : train/0/15804.jpg
Image : train/0/15806.jpg
Image : train/0/15807.jpg
Image : train/0/15809.jpg
Image : train/0/15810.jpg
Image : train/0/15811.jpg
Image : train/0/15812.jpg
Image : train/0/15814.jpg
Image : trai

Image : train/0/16451.jpg
Image : train/0/16455.jpg
Image : train/0/16456.jpg
Image : train/0/16458.jpg
Image : train/0/16468.jpg
Image : train/0/16470.jpg
Image : train/0/16471.jpg
Image : train/0/16476.jpg
Image : train/0/16478.jpg
Image : train/0/16479.jpg
Image : train/0/16483.jpg
Image : train/0/16486.jpg
Image : train/0/16494.jpg
Image : train/0/16496.jpg
Image : train/0/16498.jpg
Image : train/0/16499.jpg
Image : train/0/16503.jpg
Image : train/0/16504.jpg
Image : train/0/16505.jpg
Image : train/0/16506.jpg
Image : train/0/16508.jpg
Image : train/0/16509.jpg
Image : train/0/16510.jpg
Image : train/0/16512.jpg
Image : train/0/16513.jpg
Image : train/0/16515.jpg
Image : train/0/16516.jpg
Image : train/0/16517.jpg
Image : train/0/16519.jpg
Image : train/0/16521.jpg
Image : train/0/16522.jpg
Image : train/0/16524.jpg
Image : train/0/16525.jpg
Image : train/0/16526.jpg
Image : train/0/16527.jpg
Image : train/0/16528.jpg
Image : train/0/16529.jpg
Image : train/0/16534.jpg
Image : trai

Image : train/0/17121.jpg
Image : train/0/17122.jpg
Image : train/0/17123.jpg
Image : train/0/17124.jpg
Image : train/0/17125.jpg
Image : train/0/17126.jpg
Image : train/0/17127.jpg
Image : train/0/17133.jpg
Image : train/0/17138.jpg
Image : train/0/17141.jpg
Image : train/0/17143.jpg
Image : train/0/17145.jpg
Image : train/0/17147.jpg
Image : train/0/17148.jpg
Image : train/0/17150.jpg
Image : train/0/17151.jpg
Image : train/0/17153.jpg
Image : train/0/17154.jpg
Image : train/0/17155.jpg
Image : train/0/17156.jpg
Image : train/0/17162.jpg
Image : train/0/17165.jpg
Image : train/0/17166.jpg
Image : train/0/17167.jpg
Image : train/0/17168.jpg
Image : train/0/17169.jpg
Image : train/0/17171.jpg
Image : train/0/17173.jpg
Image : train/0/17179.jpg
Image : train/0/17182.jpg
Image : train/0/17183.jpg
Image : train/0/17184.jpg
Image : train/0/17191.jpg
Image : train/0/17192.jpg
Image : train/0/17195.jpg
Image : train/0/17200.jpg
Image : train/0/17201.jpg
Image : train/0/17202.jpg
Image : trai

Image : train/0/17791.jpg
Image : train/0/17792.jpg
Image : train/0/17795.jpg
Image : train/0/17796.jpg
Image : train/0/17797.jpg
Image : train/0/17798.jpg
Image : train/0/17799.jpg
Image : train/0/17800.jpg
Image : train/0/17801.jpg
Image : train/0/17802.jpg
Image : train/0/17803.jpg
Image : train/0/17804.jpg
Image : train/0/17808.jpg
Image : train/0/17809.jpg
Image : train/0/17812.jpg
Image : train/0/17813.jpg
Image : train/0/17814.jpg
Image : train/0/17816.jpg
Image : train/0/17819.jpg
Image : train/0/17821.jpg
Image : train/0/17822.jpg
Image : train/0/17823.jpg
Image : train/0/17824.jpg
Image : train/0/17825.jpg
Image : train/0/17827.jpg
Image : train/0/17828.jpg
Image : train/0/17829.jpg
Image : train/0/17830.jpg
Image : train/0/17832.jpg
Image : train/0/17833.jpg
Image : train/0/17834.jpg
Image : train/0/17836.jpg
Image : train/0/17837.jpg
Image : train/0/17840.jpg
Image : train/0/17841.jpg
Image : train/0/17845.jpg
Image : train/0/17846.jpg
Image : train/0/17850.jpg
Image : trai

Image : train/0/18449.jpg
Image : train/0/18451.jpg
Image : train/0/18452.jpg
Image : train/0/18454.jpg
Image : train/0/18455.jpg
Image : train/0/18456.jpg
Image : train/0/18457.jpg
Image : train/0/18458.jpg
Image : train/0/18459.jpg
Image : train/0/18460.jpg
Image : train/0/18461.jpg
Image : train/0/18462.jpg
Image : train/0/18464.jpg
Image : train/0/18465.jpg
Image : train/0/18466.jpg
Image : train/0/18468.jpg
Image : train/0/18471.jpg
Image : train/0/18475.jpg
Image : train/0/18476.jpg
Image : train/0/18477.jpg
Image : train/0/18479.jpg
Image : train/0/18482.jpg
Image : train/0/18485.jpg
Image : train/0/18488.jpg
Image : train/0/18490.jpg
Image : train/0/18495.jpg
Image : train/0/18496.jpg
Image : train/0/18498.jpg
Image : train/0/18502.jpg
Image : train/0/18503.jpg
Image : train/0/18505.jpg
Image : train/0/18506.jpg
Image : train/0/18508.jpg
Image : train/0/18509.jpg
Image : train/0/18512.jpg
Image : train/0/18513.jpg
Image : train/0/18518.jpg
Image : train/0/18521.jpg
Image : trai

Image : train/0/19191.jpg
Image : train/0/19192.jpg
Image : train/0/19195.jpg
Image : train/0/19196.jpg
Image : train/0/19198.jpg
Image : train/0/19199.jpg
Image : train/0/19200.jpg
Image : train/0/19201.jpg
Image : train/0/19206.jpg
Image : train/0/19207.jpg
Image : train/0/19209.jpg
Image : train/0/19210.jpg
Image : train/0/19211.jpg
Image : train/0/19212.jpg
Image : train/0/19213.jpg
Image : train/0/19214.jpg
Image : train/0/19215.jpg
Image : train/0/19216.jpg
Image : train/0/19218.jpg
Image : train/0/19219.jpg
Image : train/0/19221.jpg
Image : train/0/19222.jpg
Image : train/0/19226.jpg
Image : train/0/19228.jpg
Image : train/0/19232.jpg
Image : train/0/19236.jpg
Image : train/0/19242.jpg
Image : train/0/19245.jpg
Image : train/0/19247.jpg
Image : train/0/19248.jpg
Image : train/0/19249.jpg
Image : train/0/19251.jpg
Image : train/0/19252.jpg
Image : train/0/19254.jpg
Image : train/0/19255.jpg
Image : train/0/19257.jpg
Image : train/0/19258.jpg
Image : train/0/19261.jpg
Image : trai

Image : train/0/19883.jpg
Image : train/0/19885.jpg
Image : train/0/19887.jpg
Image : train/0/19889.jpg
Image : train/0/19891.jpg
Image : train/0/19893.jpg
Image : train/0/19895.jpg
Image : train/0/19896.jpg
Image : train/0/19897.jpg
Image : train/0/19898.jpg
Image : train/0/19900.jpg
Image : train/0/19902.jpg
Image : train/0/19907.jpg
Image : train/0/19909.jpg
Image : train/0/19910.jpg
Image : train/0/19911.jpg
Image : train/0/19913.jpg
Image : train/0/19914.jpg
Image : train/0/19915.jpg
Image : train/0/19916.jpg
Image : train/0/19917.jpg
Image : train/0/19918.jpg
Image : train/0/19920.jpg
Image : train/0/19924.jpg
Image : train/0/19928.jpg
Image : train/0/19932.jpg
Image : train/0/19934.jpg
Image : train/0/19936.jpg
Image : train/0/19937.jpg
Image : train/0/19938.jpg
Image : train/0/19939.jpg
Image : train/0/19940.jpg
Image : train/0/19941.jpg
Image : train/0/19942.jpg
Image : train/0/19944.jpg
Image : train/0/19946.jpg
Image : train/0/19948.jpg
Image : train/0/19950.jpg
Image : trai

Image : train/0/20476.jpg
Image : train/0/20477.jpg
Image : train/0/20478.jpg
Image : train/0/20483.jpg
Image : train/0/20484.jpg
Image : train/0/20485.jpg
Image : train/0/20488.jpg
Image : train/0/20490.jpg
Image : train/0/20496.jpg
Image : train/0/20497.jpg
Image : train/0/20498.jpg
Image : train/0/20499.jpg
Image : train/0/20501.jpg
Image : train/0/20502.jpg
Image : train/0/20504.jpg
Image : train/0/20505.jpg
Image : train/0/20507.jpg
Image : train/0/20508.jpg
Image : train/0/20509.jpg
Image : train/0/20511.jpg
Image : train/0/20514.jpg
Image : train/0/20515.jpg
Image : train/0/20516.jpg
Image : train/0/20520.jpg
Image : train/0/20521.jpg
Image : train/0/20522.jpg
Image : train/0/20523.jpg
Image : train/0/20524.jpg
Image : train/0/20526.jpg
Image : train/0/20527.jpg
Image : train/0/20530.jpg
Image : train/0/20531.jpg
Image : train/0/20532.jpg
Image : train/0/20533.jpg
Image : train/0/20537.jpg
Image : train/0/20538.jpg
Image : train/0/20539.jpg
Image : train/0/20542.jpg
Image : trai

Image : train/0/21193.jpg
Image : train/0/21197.jpg
Image : train/0/21198.jpg
Image : train/0/21202.jpg
Image : train/0/21203.jpg
Image : train/0/21205.jpg
Image : train/0/21206.jpg
Image : train/0/21207.jpg
Image : train/0/21210.jpg
Image : train/0/21215.jpg
Image : train/0/21217.jpg
Image : train/0/21218.jpg
Image : train/0/21219.jpg
Image : train/0/21220.jpg
Image : train/0/21221.jpg
Image : train/0/21222.jpg
Image : train/0/21223.jpg
Image : train/0/21224.jpg
Image : train/0/21228.jpg
Image : train/0/21229.jpg
Image : train/0/21231.jpg
Image : train/0/21233.jpg
Image : train/0/21237.jpg
Image : train/0/21238.jpg
Image : train/0/21239.jpg
Image : train/0/21244.jpg
Image : train/0/21249.jpg
Image : train/0/21250.jpg
Image : train/0/21253.jpg
Image : train/0/21254.jpg
Image : train/0/21255.jpg
Image : train/0/21256.jpg
Image : train/0/21260.jpg
Image : train/0/21261.jpg
Image : train/0/21263.jpg
Image : train/0/21266.jpg
Image : train/0/21268.jpg
Image : train/0/21271.jpg
Image : trai

## Shuffling the data and clearing the session.

In [6]:
random.shuffle(D)
random.shuffle(D)
random.shuffle(D)

len(D)

21765

## Spliting the data into test and train.

splitting the data into test and train.

In [7]:
X, y = zip(*D)
X_train, X_test, y_train, y_test = train_test_split(X, y, stratify=y, test_size = 0.1)

Reshape for CNN input.

In [8]:
X_train = np.array([x.reshape( (128, 128, 3) ) for x in X_train])
X_test = np.array([x.reshape( (128, 128, 3) ) for x in X_test])

One-Hot encoding for classes.

In [9]:
y_train = np.array(y_train)
y_test = np.array(y_test)

## Custom model definition.

In [10]:
def Model():
   
    model = Sequential()
    
    model.add(Conv2D(32, (3, 3), input_shape=(128, 128, 3)))
    model.add(Activation('relu'))
    model.add(MaxPooling2D(pool_size=(2, 2)))
    
    model.add(Conv2D(32, (3, 3)))
    model.add(Activation('relu'))
    model.add(MaxPooling2D(pool_size=(2, 2)))
    
    model.add(Conv2D(64, (3, 3)))
    model.add(Activation('relu'))
    model.add(MaxPooling2D(pool_size=(2, 2)))

    model.add(Conv2D(64, (3, 3)))
    model.add(Activation('relu'))
    model.add(MaxPooling2D(pool_size=(2, 2)))
    
    model.add(Flatten())
    
    model.add(Dense(64, activation='relu'))
    model.add(Dense(1, activation='sigmoid'))
    #model.add(Dropout(0.5))
    
    #fix the optimizer
    sgd = SGD(lr=1e-1, decay=1e-6, momentum=0.9, nesterov=True) #try varying this for your task and see the best fit
    
    #compile the model
    model.compile(optimizer='adam', loss='binary_crossentropy', metrics=['accuracy'])
    
    return model


## Creating the model { Resnet Model }.

def Model():

    base_model = applications.resnet50.ResNet50(weights = None , include_top = False, input_shape= (128, 128, 1))
    
    x = base_model.output
    
    #for layer in base_model.layers:
        #layer.trainable = False 
    
    x = GlobalAveragePooling2D()(x)
    x = Dropout(0.5)(x)
    
    predictions = Dense(number_of_class, activation= 'softmax')(x)
    model = Model(inputs = base_model.input, outputs = predictions)
    
    # sgd = SGD(lr=0.01, decay=1e-6, momentum=0.9, nesterov=True)
    model.compile(optimizer= "adam", loss='categorical_crossentropy', metrics=['accuracy'])
    
    return model


## Creating the model { Vgg16 Model }.

def Model():  
    
    base_model = applications.vgg16.VGG16(weights = 'imagenet', include_top = False, input_shape = (128, 128, 1))
    
    x = base_model.output

    #for layer in base_model.layers:
        #layer.trainable = False 

    x = TimeDistributed(Flatten())(x)
    x = LSTM(256, return_sequences=False, dropout=0.5)(x)
    
    predictions = Dense(number_of_class, activation="softmax")(x)
    model = Model(inputs = base_model.inputs, outputs=predictions)

    sgd = SGD(lr=0.00001, decay=1e-6, momentum=0.9, nesterov=True) 
    model.compile(optimizer= "adam", loss='categorical_crossentropy', metrics=['accuracy'])
    
    return model

## Defining the callback elements.

In [11]:
model = Model()
model.summary()

filepath = model.name + '.{epoch:02d}-{val_loss:.4f}-{val_acc:.4f}.h5'

# Set early stopping criteria.

# Patience is the number of epochs with no improvment after which the training will stop.
#-----------------------------------------------------------------------------------------
early_stopping = EarlyStopping(monitor='val_loss', patience=5, verbose=1)

# Define the model checkpoint callback -> this will keep on saving the model as a physical file
#----------------------------------------------------------------------------------------------
model_checkpoint = ModelCheckpoint(filepath, verbose=1, save_best_only=True)


_________________________________________________________________
Layer (type)                 Output Shape              Param #   
conv2d_1 (Conv2D)            (None, 126, 126, 32)      896       
_________________________________________________________________
activation_1 (Activation)    (None, 126, 126, 32)      0         
_________________________________________________________________
max_pooling2d_1 (MaxPooling2 (None, 63, 63, 32)        0         
_________________________________________________________________
conv2d_2 (Conv2D)            (None, 61, 61, 32)        9248      
_________________________________________________________________
activation_2 (Activation)    (None, 61, 61, 32)        0         
_________________________________________________________________
max_pooling2d_2 (MaxPooling2 (None, 30, 30, 32)        0         
_________________________________________________________________
conv2d_3 (Conv2D)            (None, 28, 28, 64)        18496     
__________

## Fitting the model. 

In [12]:
def fit_and_evaluate(t_x, val_x, t_y, val_y):
    
    model = None
    model = Model()
    
    results = model.fit(t_x, 
                        t_y, 
                        epochs=10, 
                        batch_size=32, 
                        callbacks=[early_stopping, model_checkpoint], 
                        verbose=1, 
                        validation_split = 0.1)  
    
    print("Val Score: ", model.evaluate(val_x, val_y))
    
    return results

Defining the number of folds and training the model.

In [13]:
n_folds = 20

for i in range(n_folds):
    
    print("Training on Fold: ", i+1)
    
    t_x, val_x, t_y, val_y = train_test_split(X_train, 
                                              y_train, 
                                              test_size = 0.1, 
                                              random_state = np.random.randint(1,1000, 1)[0])
    
    model_history.append(fit_and_evaluate(t_x, val_x, t_y, val_y))
    
    print("======="*12, end="\n\n\n")
    

Training on Fold:  1
Train on 15866 samples, validate on 1763 samples
Epoch 1/10
15866/15866 [==============================] - 16s 1ms/step - loss: 0.6705 - acc: 0.5811 - val_loss: 0.6221 - val_acc: 0.6483

Epoch 00001: val_loss improved from inf to 0.62211, saving model to sequential_1.01-0.6221-0.6483.h5
Epoch 2/10
15866/15866 [==============================] - 10s 627us/step - loss: 0.2838 - acc: 0.8772 - val_loss: 0.1490 - val_acc: 0.9558

Epoch 00002: val_loss improved from 0.62211 to 0.14902, saving model to sequential_1.02-0.1490-0.9558.h5
Epoch 3/10
15866/15866 [==============================] - 10s 629us/step - loss: 0.1343 - acc: 0.9547 - val_loss: 0.1466 - val_acc: 0.9495

Epoch 00003: val_loss improved from 0.14902 to 0.14660, saving model to sequential_1.03-0.1466-0.9495.h5
Epoch 4/10
15866/15866 [==============================] - 10s 629us/step - loss: 0.1215 - acc: 0.9573 - val_loss: 0.1333 - val_acc: 0.9580

Epoch 00004: val_loss improved from 0.14660 to 0.13328, savin


Epoch 00009: val_loss did not improve from 0.11803
Epoch 10/10
15866/15866 [==============================] - 10s 626us/step - loss: 0.0714 - acc: 0.9735 - val_loss: 0.1276 - val_acc: 0.9546

Epoch 00010: val_loss did not improve from 0.11803
1959/1959 [==============================] - 1s 314us/step
Val Score:  [0.16178816937826737, 0.9515058703420113]


Training on Fold:  5
Train on 15866 samples, validate on 1763 samples
Epoch 1/10
15866/15866 [==============================] - 11s 699us/step - loss: 0.2734 - acc: 0.8893 - val_loss: 0.1458 - val_acc: 0.9546

Epoch 00001: val_loss did not improve from 0.11803
Epoch 2/10
15866/15866 [==============================] - 10s 635us/step - loss: 0.1496 - acc: 0.9527 - val_loss: 0.1552 - val_acc: 0.9507

Epoch 00002: val_loss did not improve from 0.11803
Epoch 3/10
15866/15866 [==============================] - 10s 636us/step - loss: 0.1352 - acc: 0.9552 - val_loss: 0.1365 - val_acc: 0.9563

Epoch 00003: val_loss did not improve from 0.1180

MemoryError: 

## Generating the Graph.

In [ ]:
# Accuracy graph.
#------------------------------------------------------------------------------    
plt.title('Accuracies vs Epochs')
plt.figure(figsize=(20,10))

for i in range(0, n_folds):
    plt.plot(model_history[i].history['acc'], label='Training Fold '+str(i))
plt.legend()
plt.show()

# Acuuracy Train vs Validation.
#------------------------------------------------------------------------------
plt.title('Train Accuracy vs Val Accuracy')
plt.figure(figsize=(20,10))

for i in range(0, n_folds):
    plt.plot(model_history[i].history['acc'], label='Train Accuracy Fold '+str(i), color='black')
    plt.plot(model_history[i].history['val_acc'], label='Val Accuracy Fold 1'+str(i), color='black', linestyle = "dashdot")
plt.legend()
plt.show()

## Testing of the model.

In [ ]:
testing = pd.read_csv('submission.csv')  

model = load_model('sequential_1.06-0.0911-0.9745.h5')

#model.evaluate(X_test, y_test)

Writing the prediction for unseen data into the csv file. 

In [ ]:

with open('submission_format.csv', 'w') as csvFile:
        csvFile.write('{},{}'.format('filename','infected'))

for i in range(len(testing.filename)):
    
    img = image.load_img('test/'+testing.filename[i], target_size=(128,128,3), color_mode='rgb')
    img = image.img_to_array(img)
    img = img/255
    
    data = np.array([img.reshape( (128, 128, 3) )])
        
    get_data = model.predict(data)
    
    print('written: '+ testing.filename[i])
    
    with open('submission_format.csv', 'a') as csvFile:
        csvFile.write('\n{},{}'.format(testing.filename[i], get_data[0][0]))

from keras.preprocessing.image import ImageDataGenerator 
from keras.models import Sequential 
from keras.layers import Conv2D, MaxPooling2D 
from keras.layers import Activation, Dropout, Flatten, Dense 
from keras import backend as K 
from keras.optimizers import SGD  
    
img_width, img_height = 128, 128
  
train_data_dir = 'v_data/train'
validation_data_dir = 'v_data/test'
nb_train_samples = 17413
nb_validation_samples = 4352

epochs = 40
batch_size = 32
  
if K.image_data_format() == 'channels_first': 
    input_shape = (3, img_width, img_height) 
else: 
    input_shape = (img_width, img_height, 3) 

model = Sequential() 
model.add(Conv2D(32, (2, 2), input_shape = input_shape)) 
model.add(Activation('relu')) 
model.add(MaxPooling2D(pool_size =(2, 2))) 
  
model.add(Conv2D(32, (2, 2))) 
model.add(Activation('relu')) 
model.add(MaxPooling2D(pool_size =(2, 2))) 
  
model.add(Conv2D(64, (2, 2))) 
model.add(Activation('relu')) 
model.add(MaxPooling2D(pool_size =(2, 2))) 
  
model.add(Flatten()) 
model.add(Dense(64)) 
model.add(Activation('relu')) 
model.add(Dropout(0.5)) 
model.add(Dense(1)) 
model.add(Activation('sigmoid')) 

sgd = SGD(lr=1e-6, decay=1e-6, momentum=0.9, nesterov=True) #try varying this for your task and see the best fit
  
model.compile(loss ='binary_crossentropy', 
                     optimizer = 'adam', 
                   metrics =['accuracy']) 
  
train_datagen = ImageDataGenerator( rescale = 1. / 255, 
                            vertical_flip = True,
                            horizontal_flip = True) 
  
test_datagen = ImageDataGenerator(rescale = 1. / 255) 
  
train_generator = train_datagen.flow_from_directory(train_data_dir, 
                                                    target_size =(img_width, img_height), 
                                                    batch_size = batch_size, class_mode ='binary') 
  
validation_generator = test_datagen.flow_from_directory(validation_data_dir, 
                                                        target_size =(img_width, img_height), 
                                                        batch_size = batch_size, class_mode ='binary') 
  
model.fit_generator(train_generator, 
                    steps_per_epoch = nb_train_samples // batch_size, 
                    epochs = epochs, validation_data = validation_generator, 
                    validation_steps = nb_validation_samples // batch_size) 
  
model.save_weights('model_saved.h5') 